neutrino.go

```diff
@@ -1251,16 +1251,14 @@ func (s *ChainService) handleDonePeerMsg(state *peerState, sp *ServerPeer) {
 		list = state.outboundPeers
 	}
 	if _, ok := list[sp.ID()]; ok {
-		if sp.VersionKnown() {
-			state.outboundGroups[addrmgr.GroupKey(sp.NA())]--
-		}
+		state.outboundGroups[addrmgr.GroupKey(sp.NA())]--
+		delete(list, sp.ID())
```

Comment:
this method really needs some comments

Reply:
Any fundamental reason to move the `delete` call here? Just grouping of related map operations I guess?

Reply:
> Just grouping of related map operations I guess?

Yeah.

> this method really needs some comments

Done.